File: core/src/main/java/net/particify/arsnova/core/web/exceptions/ForbiddenException.java
package net.particify.arsnova.core.web.exceptions;

import java.io.Serial;

/**
 * Forbidden means status code 403.
 */
public class ForbiddenException extends RuntimeException {
  @Serial
  private static final long serialVersionUID = 1L;

  public ForbiddenException() {
    super();
  }

  public ForbiddenException(final String message) {
    super(message);
  }

  public ForbiddenException(final Throwable e) {
    super(e);
  }

  public ForbiddenException(final String message, final Throwable e) {
    super(message, e);
  }
}

File: gateway/src/main/kotlin/net/particify/arsnova/gateway/exception/ForbiddenException.kt
package net.particify.arsnova.gateway.exception

import org.springframework.http.HttpStatus
import org.springframework.web.server.ResponseStatusException

class ForbiddenException : ResponseStatusException {
  constructor() : super(HttpStatus.FORBIDDEN)
  constructor(reason: String?) : super(HttpStatus.FORBIDDEN, reason)
  constructor(reason: String?, cause: Throwable?) : super(HttpStatus.FORBIDDEN, reason, cause)
}
